src/uu/seq/src/seq.rs

```diff
@@ -230,54 +128,33 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
         );
         return 1;
     }
-    let last: Number = {
+    let last: PreciseNumber = {
         let slice = numbers[numbers.len() - 1];
-        let int_digits = num_integral_digits(slice).unwrap_or_else(|_| {
-            crash!(
-                1,
-                "invalid floating point argument: {}\n Try '{} --help' for more information.",
-                slice.quote(),
-                uucore::execution_phrase()
-            )
-        });
-        padding = cmp::max(padding, int_digits);
-        crash_if_err!(1, slice.parse())
+        slice.parse().unwrap_or_else(|e| exit_with_error(slice, e))
     };
 
-    let is_negative_zero_f64 = |x: f64| x == -0.0 && x.is_sign_negative() && largest_dec == 0;
-    let result = match (first, last, increment) {
-        // For example, `seq -0 1 2` or `seq -0 1 2.0`.
-        (Number::MinusZero, last, Number::BigInt(increment)) => print_seq_integers(
-            (BigInt::zero(), increment, last.into_bigint()),
-            options.separator,
-            options.terminator,
-            options.widths,
-            padding,
-            true,
-        ),
-        // For example, `seq -0e0 1 2` or `seq -0e0 1 2.0`.
-        (Number::F64(x), last, Number::BigInt(increment)) if is_negative_zero_f64(x) => {
-            print_seq_integers(
-                (BigInt::zero(), increment, last.into_bigint()),
-                options.separator,
-                options.terminator,
-                options.widths,
-                padding,
-                true,
-            )
-        }
-        // For example, `seq 0 1 2` or `seq 0 1 2.0`.
-        (Number::BigInt(first), last, Number::BigInt(increment)) => print_seq_integers(
-            (first, increment, last.into_bigint()),
+    let padding = first
+        .num_integral_digits
+        .max(increment.num_integral_digits)
+        .max(last.num_integral_digits);
+    let largest_dec = first
+        .num_fractional_digits
+        .max(increment.num_fractional_digits);
+
+    let result = match (first.number, increment.number, last.number) {
+        (Number::Int(first), Number::Int(increment), last) => print_seq_integers(
+            (first, increment, last.into_extended_big_int()),
```

Comment:
`last.into_extended_big_int()` (i.e. truncating the BigDecimal) is not always the right thing to do. Consider `seq 5 -1 2.2`: `2.2` is truncated to `2`, so `2` is printed as well. I think we should instead round `last` towards `first`.

Reply:
I'll add that test case, thanks.